### PR TITLE
Use SystemTableProvider in SystemTableAwareAccessControl

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 1]
```diff
@@ -3008,11 +3007,12 @@ public boolean delegateMaterializedViewRefreshToConnector(ConnectorSession sessi
         return hiveMaterializedViewMetadata.refreshMaterializedView(session, name);
     }
 
-    public static Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
+    private Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
     {
-        return Stream.of(SystemTableHandler.values())
-                .filter(handler -> handler.matches(tableName))
-                .map(handler -> handler.getSourceTableName(tableName))
+        return systemTableProviders.stream()
```
Comment (losipiuk) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:3012]: Are there any other uses of `SystemTableHandler.values`?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SystemTableAwareAccessControl.java [hunk 2]
```diff
@@ -87,4 +92,13 @@ public void checkCanSelectFromColumns(ConnectorSecurityContext context, SchemaTa
 
         delegate.checkCanSelectFromColumns(context, tableName, columnNames);
     }
+
+    private Optional<SchemaTableName> getSourceTableNameFromSystemTable(SchemaTableName tableName)
```
Comment (losipiuk) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SystemTableAwareAccessControl.java:96]: can we not copy-paste logic?

Comment (losipiuk) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SystemTableAwareAccessControl.java:96]: You at very least can keep the static method, just parameterized it with `Set<SystemTableProvider>`. And call in two places.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 1 above)
Comment (sopel39) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:3012]: no

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SystemTableAwareAccessControl.java (hunk 2 above)
Comment (sopel39) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/SystemTableAwareAccessControl.java:96]: done

**Review by losipiuk — APPROVED** [review]

**Pull request merged**